Add ownership verification steps to connection flow

delputnam opened the pull request:
#### Changes proposed in this Pull Request

In certain cases, we may want to request that users verify ownership of domains they want to connect to their site. This PR adds a flow to allow this when connecting a domain to a site with a paid plan.

When you are on a site that requires ownership verification, you will first be prompted to log in to your provider:
<img width="1792" alt="Screen Shot 2021-09-09 at 10 07 33 AM" src="https://user-images.githubusercontent.com/1379730/132701784-bc10d557-e035-4240-b0a7-6659e668dc9b.png">


In the next step, you will be asked to enter the auth code for your domain:
<img width="1792" alt="Screen Shot 2021-09-09 at 10 07 39 AM" src="https://user-images.githubusercontent.com/1379730/132701828-7f477d5d-df0d-4e6b-a9de-5aea9c5c8cf1.png">


Leaving the auth code field blank or entering an incorrect value will cause errors to be shown below the auth code field:
<img width="1792" alt="Screen Shot 2021-09-09 at 10 07 44 AM" src="https://user-images.githubusercontent.com/1379730/132701932-cc09e66c-40d7-493f-af57-8fd413554a9a.png">


When the correct auth code is entered, the domain will be connected and the user will be directed to the "connect a domain flow" as usual. 

#### Testing instructions

Hack the back end to return `{ ownership_verification_type: 'auth_code' }` on the `is-available` endpoint.

Select a site that is on a paid plan and go through the "Use a domain I own" to connect a domain you already own that is not mapped to WordPress.com.

Try to verify the auth code without entering anything in the input field.
Try entering the incorrect auth code in the input field.
Make sure that your domain is properly connected when you enter the correct auth code.

delputnam committed:
Add ownership verification steps to connection flow

a8ci18n commented on client/components/domains/connect-domain-step/connect-domain-step-ownership-auth-code.jsx:
```diff
@@ -0,0 +1,141 @@
+				onClick={ checkAuthCode }
+				primary
+			>
+				{ __( 'Check authorization code' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Check my authorization code' )` *ES Score: 9*

delputnam committed:
Update header

poligilad-auto reviewed (dismissed):
Looking good @delputnam.

1. The first step is missing some copy. I added in an additional sentence before the explanation about how to log in, to explain why they need to go through this step. "We need to confirm that you are authorized to connect this domain." ([Figma file](https://www.figma.com/file/kjzWtItHG8sIJZBJGxpWBX/Early-touch-domains?node-id=2104%3A74548))
2. For the second step, following your comment we can use "Enter authorization code" as the placeholder text and "Check my authorization code" as the CTA.

hambai commented on packages/wpcom.js/src/lib/domains.js:
```diff
@@ -63,6 +63,19 @@ class Domains {
+	 * @param {Function} fn The callback function
+	 * @returns {Promise} A promise that resolves when the request completes
+	 */
+	authCodeCheck( domain, authCode, fn ) {
```
We already have a method called `checkAuthCode` in undocumented.js

rafaelgallani reviewed (commented):
Nice, LGTM! After merging #56031, this one should be good to go code-wise 😬

delputnam committed:
Update text on auth code check page

delputnam committed:
Ensure that isOwnershipVerificationFlow is set

delputnam commented:
I've addressed @poligilad-auto 's comments in https://github.com/Automattic/wp-calypso/pull/56011/commits/7151e52123d916a932d7dc9d21d37da5eb16879f and https://github.com/Automattic/wp-calypso/pull/56011/commits/25a2d0ed4eaff99d56eba47b7add3422d545260c

hambai committed:
Update the domain mapping verification form to only call the add mapping endpoint and show the error that is returned from it (#56031)

In order to simplify the domain mapping verification I believe we should call only the add domain mapping endpoint with all the data provided. That way we won't check the auth code more than once and also this'll probably simplify future ownership verification types.

delputnam committed:
Only show ownership verification errors inline

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6578831

Thank you @delputnam for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6578831).
